src/posix/platform/trel_udp6.cpp

```diff
@@ -751,6 +900,23 @@ void platformTrelProcess(otInstance *aInstance, const fd_set *aReadFdSet, const
         ReceivePacket(sMulticastSocket, aInstance);
     }
 
+    if (FD_ISSET(sNetlinkSocket, aReadFdSet))
+    {
+        ReceiveNetlinkMessage();
+    }
+
+    if (sAddInterfaceAddressDeadline > 0)
+    {
+        uint64_t now = otPlatTimeGet();
+
+        if (sAddInterfaceAddressDeadline <= now)
+        {
+            sAddInterfaceAddressDeadline = 0;
+
+            HandleAddInterfaceAddressTimeout();
```

Comment:
Do we need to move this and the checks above under `#if OPENTHREAD_CONFIG_POSIX_TREL_USE_NETLINK_SOCKET`?

I think the `HandleAddInterfaceAddressTimeout()` is now only defined under this config above, right?

Reply:
I will make sure everything works in both configurations.